pkg/api/v1/object-storage/buckets/bucket.go

```diff
@@ -86,17 +87,29 @@ func (svc *ObjectStorageBucketsService) Create(ctx context.Context, projectID, b
 		return
 	}
 
-	_, err = svc.Client.Do(req, nil)
+	w = wait.New(func() (interface{}, bool, error) {
```

Comment:
to make the code prettier, I've separated the waitFn definition to a separate function. would be great if we can keep it consistent

the naming convention so far for those functions are `waitForCreation` and `waitForDeletion`